src/Appwrite/Specification/Format.php

```diff
@@ -97,4 +98,106 @@ public function getParam(string $key, string $default = ''): string
     {
         return $this->params[$key] ?? $default;
     }
+
+    protected function getEnumName(string $service, string $method, string $param): ?string
+    {
+        switch ($service) {
+            case 'account':
+                switch ($method) {
+                    case 'createOAuth2Session':
+                        return 'Provider';
+                }
+                break;
+            case 'avatars':
+                switch ($method) {
+                    case 'getBrowser':
+                        return 'Browser';
+                    case 'getCreditCard':
+                        return 'CreditCard';
+                    case 'getFlag':
+                        return  'Flag';
+                }
+                break;
+            case 'storage':
+                switch ($method) {
+                    case 'getFilePreview':
+                        switch ($param) {
+                            case 'gravity':
+                                return 'ImageGravity';
+                            case 'output':
+                                return  'ImageFormat';
+                        }
+                        break;
+                }
+                break;
+        }
```

Comment:
We're missing functions from the database, functions, projects and users services here

Reply:
They don't have enum field set in the spec file (https://raw.githubusercontent.com/appwrite/appwrite/faece71145f680a839ec70d41e6eefe9e341fdbe/app/config/specs/swagger2-latest-console.json).

Is that expected or we are missing something?

Reply:
We're missing something 🤔 every parameter that uses the WhiteList validator should have the enum property. A project search for `new WhiteList` should show all the routes that need added